src/starkware_utils/commitment_tree/binary_fact_tree_node.rs

```diff
@@ -108,9 +109,10 @@ where
 
     async fn get_children(&mut self, node: &Self::NodeType) -> Result<Vec<Self::NodeType>, TreeError> {
         if node.previous.is_leaf() {
+            let storage = self.ffc.storage().await;
             // unwrap() on leaf_hash is guaranteed to be safe because of the check above
-            let previous = LF::get_or_fail(&self.ffc.storage, &node.previous.leaf_hash().unwrap()).await?;
-            let current = LF::get_or_fail(&self.ffc.storage, &node.current.leaf_hash().unwrap()).await?;
+            let previous = LF::get_or_fail(storage.deref(), &node.previous.leaf_hash().unwrap()).await?;
+            let current = LF::get_or_fail(storage.deref(), &node.current.leaf_hash().unwrap()).await?;
```

Comment:
```suggestion
            let previous = LF::get_or_fail(&*storage, &node.previous.leaf_hash().unwrap()).await?;
            let current = LF::get_or_fail(&*storage, &node.current.leaf_hash().unwrap()).await?;
```

Maybe, maybe not...

Reply:
Yeah the `&*` pattern feels ugly to me, I prefer an explicit `deref()`.